lib/src/signature_util.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(missing_docs)]
+
+use crate::backend::{MillisSinceEpoch, Signature, Timestamp};
+
+const EMPTY_STRING_PLACEHOLDER: &str = "JJ_EMPTY_STRING";
```

Comment:
iirc, this placeholder value is a workaround specific to the git backend, which can't store an empty string. It shouldn't apply to the default (and maybe Google) backends.

Reply:
Right, that makes sense. In this case it would only be relevant if you did something very weird like put `Nobody At All <JJ_EMPTY_STRING>` in the `.mailmap` file. Maybe I could add a special case to the `.mailmap` code to handle that? Or this part of the conversion could be factored out to be directly in the Git backend instead.

Reply:
I could also just revert this code migration and hand‐code the conversion in the `mailmap` module. Without timestamp conversion or empty name handling it’s very brief so the remaining duplication would be fine, and it’s less code churn.

OTOH that would make the results inconsistent with Git when you are using the Git backend and try to put the empty user in `.mailmap`, but… well… probably don’t do that?

Reply:
I think our `Mailmap::resolve()` can just return early if the signature is empty. It's invalid in Git, and so empty name in mailmap would also be invalid.

Alternatively, maybe we can just reuse the parser part and do lookup by ourselves?

Reply:
Returning early sounds reasonable to me. It will result in inconsistency with Git if anyone puts `<JJ_EMPTY_STRING>` maps in their `.mailmap`, but I don’t think that divergence matters considering how pathological a case that is.